### PR TITLE
HPCC-27004 Split Esp DFS service into it's own service

### PR DESCRIPTION
**Pull request by jakesmith**
For redundancy, updates etc., split the ESP DFS service into
it's own service deployment (i.e. out of eclwatch).

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
NB: don't merge / will rebase, once HPCC-27013 is merged (currently 1st commit of this PR).

**Comment by jakesmith** [comment]
@afishbeck - please review (2nd commit), 1st commit is another dependent PR (HPCC-27013)

File: esp/applications/dfs/ldap_authorization_map.yaml
Comment (afishbeck) [inline esp/applications/dfs/ldap_authorization_map.yaml:5]: @jakesmith if it's needed, the auth resource map now goes here.

```
   resource_map:
      ws_dfsservice:
         Feature:
         -  path: WsDfsAccess
            resource: WsDfsAccess
            description: Base access to DFS services
```

Comment (afishbeck) [reply esp/applications/dfs/ldap_authorization_map.yaml:5]: (edited to match the auth feature tag used in your esdl ecm file.

Comment (afishbeck) [reply esp/applications/dfs/ldap_authorization_map.yaml:5]: Also I just noticed that in the environment xml related changes for the baremetal deployment I think you used the auth feature tag "WsDfsAccess".  The name has to match what you reference in anywhere in your esdl ecm file, or c++ code.

In this case:

```
auth_feature("DfuAccess:READ")
```

Comment (jakesmith) [reply esp/applications/dfs/ldap_authorization_map.yaml:5]: yes, ok, will need correcting in other PR

Comment (jakesmith) [reply esp/applications/dfs/ldap_authorization_map.yaml:5]: See : https://github.com/hpcc-systems/HPCC-Platform/pull/15686

Comment (jakesmith) [reply esp/applications/dfs/ldap_authorization_map.yaml:5]: > @jakesmith if it's needed, the auth resource map now goes here.
> 
> ```
>    resource_map:
>       ws_dfsservice:
>          Feature:
>          -  path: DfsAccess
>             resource: DfsAccess
>             description: Base access to DFS services
> ```

Will correct this once https://github.com/hpcc-systems/HPCC-Platform/pull/15686 is merged

**Review by afishbeck — APPROVED** [review]

**Pull request merged**
